flink-contrib/src/test/java/org/apache/flink/contrib/operatorstatistics/OperatorStatsAccumulatorsTest.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.api.java.io.DiscardingOutputFormat;
+import org.apache.flink.api.java.tuple.Tuple1;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.test.util.AbstractTestBase;
+import org.apache.flink.util.Collector;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Random;
+
+public class OperatorStatsAccumulatorsTest extends AbstractTestBase {
+
+	private static final Logger LOG = LoggerFactory.getLogger(OperatorStatsAccumulatorsTest.class);
+
+	private static final String ACCUMULATOR_NAME = "op-stats";
+
+	public OperatorStatsAccumulatorsTest(){
+		super(new Configuration());
+	}
+
+	@Test
+	public void testAccumulator() throws Exception {
+
+		String input = "";
+
+		Random rand = new Random();
+
+		for (int i = 1; i < 1000; i++) {
+			if(rand.nextDouble()<0.2){
+				input+=String.valueOf(rand.nextInt(5))+"\n";
+			}else{
+				input+=String.valueOf(rand.nextInt(100))+"\n";
+			}
+		}
+
+		String inputFile = createTempFile("datapoints.txt", input);
+
+		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+		env.readTextFile(inputFile).
+				flatMap(new StringToInt()).
+				output(new DiscardingOutputFormat<Tuple1<Integer>>());
+
+		JobExecutionResult result = env.execute();
+
+		OperatorStatistics globalStats = result.getAccumulatorResult(ACCUMULATOR_NAME);
+		LOG.debug("Global Stats");
+		LOG.debug(globalStats.toString());
+
+		OperatorStatistics merged = null;
+
+		Map<String,Object> accResults = result.getAllAccumulatorResults();
+		for (String accumulatorName:accResults.keySet()){
+			if (accumulatorName.contains(ACCUMULATOR_NAME+"-")){
+				OperatorStatistics localStats = (OperatorStatistics) accResults.get(accumulatorName);
+				if (merged == null){
+					merged = localStats.clone();
+				}else {
+					merged.merge(localStats);
+				}
+				LOG.debug("Local Stats: " + accumulatorName);
+				LOG.debug(localStats.toString());
+			}
+		}
+
+		Assert.assertEquals(globalStats.cardinality,999);
+		Assert.assertEquals(globalStats.estimateCountDistinct(),100);
+		Assert.assertTrue(globalStats.getHeavyHitters().size()>0 && globalStats.getHeavyHitters().size()<=5);
+		Assert.assertEquals(merged.getMin(),globalStats.getMin());
+		Assert.assertEquals(merged.getMax(),globalStats.getMax());
+		Assert.assertEquals(merged.estimateCountDistinct(),globalStats.estimateCountDistinct());
+		Assert.assertEquals(merged.getHeavyHitters().size(),globalStats.getHeavyHitters().size());
+
+	}
+
+	public static class StringToInt extends RichFlatMapFunction<String, Tuple1<Integer>> {
+
+		// Is instantiated later since the runtime context is not yet initialized
+		private Accumulator<Object, Serializable> globalAccumulator;
+		private Accumulator<Object,Serializable>[] localAccumulators;
```

Comment:
This is the solution I have found for temporarily storing local, as well as global accumulators. But I think there should be a better way to do this, rather than having the user have to specify it and later update both inside of the UDF.